src/Testing/src/xunit/SkipOnAlpineAttribute.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Testing
+{
+    /// <summary>
+    /// Skip test if running on Alpine Linux (which uses musl instead of glibc)
+    /// </summary>
+    public class SkipOnAlpineAttribute : Attribute, ITestCondition
+    {
+        public SkipOnAlpineAttribute(string issueUrl = "")
+        {
+            IssueUrl = issueUrl;
+        }
+
+        public string IssueUrl { get; }
+
+        public bool IsMet { get; } = !IsAlpine;
+
+        public string SkipReason => "Test cannot run on Alpine Linux.";
+
+        // This logic is borrowed from https://github.com/dotnet/runtime/blob/6a5a78bec9a6e14b4aa52cd5ac558f6cf5c6a211/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs
+        private static bool IsAlpine { get; } =
+            RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && File.Exists("/etc/os-release") &&
+            File.ReadAllLines("/etc/os-release").Any(line =>
+                line.StartsWith("ID=", StringComparison.Ordinal) && line.Substring(3).Trim('"', '\'') == "alpine");
```

Comment:
Why can't this logic be included in `[SkipOnHelix]` to support `All.Alpine`❔ The inconsistency and the need for yet-another attribute are jarring.

Reply:
Is it because its not specific to helix only?

Reply:
Unlike the base queue name, the name of the docker image doesn't show up in the "helix" environment variable `[SkipOnHelix]` uses so @MattGal suggested this. I prefer this anyway because it does the right thing even if someone is manually running tests on Alpine.

Reply:
We actually control setting the helix variable, but right in this case the queue name is a super unwieldy docker image :(  https://github.com/dotnet/aspnetcore/blob/master/eng/helix/content/RunTests/TestRunner.cs#L37

Reply:
Either way, I think this is better because it works even when not being run using a helix queue.

Reply:
If that's compelling (I'm not 💯% convinced but could be persuaded), we need to separate `All.OSX` from `[SkipOnHelix]` and put platform testing logic into a new `[SkipOnOSX]` or `[SkipOnMacOS]` attribute.

If we continue to create distinct test conventions, it'll keep getting harder to grasp how skip tests and avoid platform-specific aka queue-specific issues.

Reply:
Well if we are going that route, we probably should add an additional environment variable which is closer to a 'tag', so for OSX queues, it could say "helix;osx;linux" the alpine queue could be "helix;alpine;linux;docker" and then we could have something like `SkipOnTag(tag = "alpine")`, `SkipOnTag(tag = "osx")`

Reply:
We'd need to add/flow that metadata in the helix definitions too, I can do that in a separate PR I guess, I don't think @halter73 should need to do that as part of this PR.

Reply:
One way to describe my feeling here is holding out for a confirmed consistent direction on how we skip for specific platforms. Bringing everything, including the existing `All.OSX` and `[OSSkipCondition]` approaches together sounds like a great direction❕ Bonus points if you subsume `[MinimumOSVersion]` and `[MaximumOSVersion]` too 😺 

I'm :ok: with follow-up work in this direction as long as we agree and have some hope it'll be done soon. Any objection @halter73❔

Reply:
Actually, why isn't this logic part of `[OSSkipCondition]`❔ Are we unable to extend the `OperatingSystems` to include narrower options❔

Reply:
If this should be added to any existing attribute, it should be added to `[OSSkipCondition]`, but that only supports specifying "Linux", "MacOSX", "Windows". I don't know why `[SkipOnHelix]` has special `All.OSX` support given that `[OSSkipCondition]` exists, but that's not really relevant to this PR.

I didn't want to add full-blown distro support to `[OSSkipCondition]` since that could get pretty involved (see runtime's [PlatformDetection.Unix.cs](https://github.com/dotnet/runtime/blob/6a5a78bec9a6e14b4aa52cd5ac558f6cf5c6a211/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs) for a taste), and I don't see us skipping any specific Linux distro other than Alpine. The only reason Alpine is special is because it uses musl instead of glibc like everey other distro. If it was easier to detect musl, I would do that.

@dougbu I need you to approve or remove your review before I can merge.

Reply:
> I need you to approve or remove your review before I can merge.

Yes, I know. Waiting to hear a confirmation @HaoK is willing to burn `All.OSX` with fire. If that happens and we can just use `[OSSkipCondition]` instead, I'll be happy because it keeps the concept count manageable.

We may still need to do some "stuff" for ARM / ARM64 skipping to make it consistent too but that's probably not going to fit into `[OSSkipCondition]` unless we add sub-categories like `Linux / Alpine` or `Windows / ARM64`.

Reply:
This PR has nothing to do with OSX. I don't think it's worthwhile to combine `[SkipOnAlpine]` with any other attribute. There's only one existing test that needed this. I don't expect it to come up much.

Keeping the concept count manageable is a good goal, but this PR does not really increase the concept count in any appreciable way.

Reply:
So that complexity is there because not everything runs on helix, so OSSkip would disable it on non helix and helix, so skip on helix is basically an additional but orthogonal slice, since we had tests that only are skipped on helix + OSX for example, that were enabled on non helix OSX.  But now that almost all tests are on helix, we can probably just use OSSkip

Reply:
What I was proposing was closer to two concepts, we live OSSkip mostly alone, and modify SkipOnHelix to have traits so you can skip on things like osx/windows/arm.  There will be some overlap with OSSkip but the tags are specific to helix queues, which is something we need to be able to target (skipping on specific queues/types of queues)

Reply:
> There's only one existing test that needed this

Then why not just use this logic in the test itself❔

We may need to fix a general issue about skipping on specific fully-qualified queue names i.e. check against the un-split name. But, I'm less convinced now that we need a new attribute at all.

> What I was proposing was closer to two concepts

I'm not understanding how developers would choose between `[OSSkipCondition]` and `[SkipOnHelix(All.OSX)]` or your proposed `[SkipOnTag(tag: "osx")]`. Why do we need two ways to skip tests on macOS, Linux, or Windows❔

Put another way, why do we need to be able to skip these general categories of Helix queues❔

Reply:
OSSkipCondition is overly broad, generally tests seem to be failing on one queue at a time, like OSSkipLinux is probably not the right thing to do when its failing only on debian, or osx, or alpine for example

Reply:
Specifically OSSkip should be used if the test will never work on OSX, SkipOnHelix("osx") should be if it works on azdo runs but doesn't work on helix

Reply:
> Specifically OSSkip should be used if the test will never work on OSX, SkipOnHelix("osx") should be if it works on azdo runs but doesn't work on helix

I get that but are any tests in the "build agent but not Helix agent" category these days❔